draft-levy-wimse-headless-jwt-authentication-practices.md

```diff
@@ -259,25 +264,58 @@ the interaction with the Authorization Server.
 
 # Interoperability Considerations {#interoperability-considerations}
 
-TODO Speak to the issue of registration/configuration, because this seems to be
-getting to the heart of the problem. The Authorization Server has a mechanism to
-get the issuer key, but how do we establish trust? Right now it seems to always
-be a manual process. [RFC7591] and [OIDC.Dynamic] are popular things to say, but
-they don't seem to be what people implement.
+In order for the workload to access the resource,
+
+1. The JWT Issuer must be recognized by the Authorization Server,
+2. Claims in the JWT are inspected and used to determine the subject, or
+   principal, of the access token issued for the Resource Server,
+3. And the resulting Resource Server principal must be authorized to access the
+   Resource.
+
+Step \#1 requires the configuration of an explicit trust relationship between
+the Authorization Server and the JWT Issuer. Despite previous attempts to
+standardize dynamic client registration in [RFC7591] and [OIDC.Dynamic], in
+practice this trust relationship, still depends on vendor-specific
+configuration.
+
+Step \#2 is a processing rule that is also previously-configured in an
+implementation-dependent manner. As an example of current practice for
+configuration of Steps \#2 and \#3, see [GitHub].
 
 # Security Considerations {#security-considerations}
 
+This document illustrates a common pattern in trust domain federation. However,
+the "identity exchange" Step \#2 in {{interoperability-considerations}} is not
+standardized. In practice, the workload platform and the Resource Server
+platform define principals differently, and the translation mechanism between
+the two identities is implemented differently by each Resource Server platform.
+This lack of standardization is not merely inconvenient; it is a rich source of
+privilege escalation attacks. This is particularly true when both the workload
+platform and the Resource Server platform are multi-tenanted.
+
+The following recommendations apply to configurations that control
+the "identity exchange" step that controls the translation of the workload JWT to a
+Resource Server identity:
+
+1. The configuration SHOULD rely on a JWT issuing key bound to a single tenant
+   of the workload platform, rather than a platform-wide JWT issuing key.
+2. The configuration SHOULD use specific JWT claims to prevent any JWT signed by
+   the JWT Issuer from being used to impersonate any Resource Server principal.
+3. The configuration SHOULD NOT rely on JWT claims that can be controlled by an
```

Comment:
"by an attacker" may create false certainty as one could be thinking "my workload is not an attacker".

I think rephrasing it to "MUST NOT solely rely on JWT claims controlled by the workload" would catch that.

Reply:
Thinking about it more, if I understand correctly you also want to cover the situation where an attacker mimics certain claims (repository name without org) to obtain access.

Maybe "identifiers that are not globally unique"?